src/generic/datavgen.cpp

```diff
@@ -1980,6 +1990,11 @@ wxDataViewMainWindow::~wxDataViewMainWindow()
 {
     DestroyTree();
     delete m_renameTimer;
+    if (GetOwner()->HasFlag(wxDV_VARIABLE_LINE_HEIGHT))
```

Comment:
I'd check `m_rowHeightCache` directly. We know that it needs to be deleted if it had been allocated, even if `wxDV_VAR_LINE_HEIGHT` flag is somehow not set any longer.

Note that the same comment applies to all the other tests of this flag in the code below, testing directly for `m_rowHeightCache` is both simpler, shorter and more robust.

Reply:
done